libevmasm/ExpressionClasses.cpp

```diff
@@ -366,21 +192,17 @@ ExpressionClasses::Id ExpressionClasses::tryToSimplify(Expression const& _expr,
 	)
 		return -1;
 
-	for (auto const& rule: rules.rules())
+	if (auto match = rules.findFirstMatch(_expr, *this))
 	{
-		rules.resetMatchGroups();
-		if (rule.first.matches(_expr, *this))
-		{
-			// Debug info
-			//cout << "Simplifying " << *_expr.item << "(";
-			//for (Id arg: _expr.arguments)
-			//	cout << fullDAGToString(arg) << ", ";
-			//cout << ")" << endl;
-			//cout << "with rule " << rule.first.toString() << endl;
-			//ExpressionTemplate t(rule.second());
-			//cout << "to " << rule.second().toString() << endl;
-			return rebuildExpression(ExpressionTemplate(rule.second(), _expr.item->location()));
-		}
+		// Debug info
+		//cout << "Simplifying " << *_expr.item << "(";
```

Comment:
This is useful code to debug optimizer problems. I can put it into an extra function, but I'm not sure it is worth it.